pyro/poutine/handlers.py

```diff
@@ -104,7 +104,7 @@ def handler(fn=None, *args, **kwargs):
             raise ValueError(
                 "{} is not callable, did you mean to pass it as a keyword arg?".format(fn))
         msngr = msngr_cls(*args, **kwargs)
-        return msngr(fn) if fn is not None else msngr
+        return functools.update_wrapper(msngr(fn), fn) if fn is not None else msngr
```

Comment:
Without this, methods that use poutines as decorators will have their docstring reset. cc @eb8680 - does this seem fine?

Reply:
LGTM

Reply:
Hmm..seems like this is breaking something - `make doctest` and `tests.poutine.test_poutines.test_decorator_interface_primitives` seem to fail. I'll investigate and update soon.